JavaFXUserInterface/src/main/java/SystemTypeInfoController.java

```diff
@@ -60,32 +62,111 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
-        //Attach the asset type delete function
         infoDeleteBtn.setOnMouseClicked(this::deleteDialog);
+
+        infoSaveBtn.setDisable(true);
+        infoSaveBtn.setOnMouseClicked(mouseEvent -> {
+            assetTypeDAO.updateAssetType(assetType.toAssetType());
+            uiUtilities.changeScene(mouseEvent, "/SystemTypeList");
+        });
+
+        systemTypeName.textProperty().addListener((obs, oldText, newText) -> {
+            if( handleTextChange(newText,originalAssetType.getName()))
+                assetType.getAssetType().setName(newText);
+        });
+        systemTypeDesc.textProperty().addListener((obs, oldText, newText) -> {
+            if( handleTextChange(newText,originalAssetType.getDescription()))
+                assetType.getAssetType().setDescription(newText);
+        });
+
+        thresholdOK.textProperty().addListener((obs, oldText, newText) -> {
+            if(handleTextChange(newText, originalAssetType.getValueOk()))
+                assetType.setValueOk(newText);
+        });
+        thresholdOK.setTextFormatter( new TextFormatter<>(c -> UIUtilities.checkFormat(TextConstants.ThresholdValueFormat,c)));
+
+        thresholdAdvisory.textProperty().addListener((obs, oldText, newText) -> {
+            if(handleTextChange(newText, originalAssetType.getValueAdvisory()))
+                assetType.setValueAdvisory(newText);
+        });
+        thresholdAdvisory.setTextFormatter( new TextFormatter<>(c -> UIUtilities.checkFormat(TextConstants.ThresholdValueFormat,c)));
+
+        thresholdCaution.textProperty().addListener((obs, oldText, newText) -> {
+            if(handleTextChange(newText, originalAssetType.getValueCaution()))
+                assetType.setValueCaution(newText);
+        });
+        thresholdCaution.setTextFormatter( new TextFormatter<>(c -> UIUtilities.checkFormat(TextConstants.ThresholdValueFormat,c)));
+
+        thresholdWarning.textProperty().addListener((obs, oldText, newText) -> {
+            if(handleTextChange(newText, originalAssetType.getValueWarning()))
+                assetType.setValueWarning(newText);
+        });
+        thresholdWarning.setTextFormatter( new TextFormatter<>(c -> UIUtilities.checkFormat(TextConstants.ThresholdValueFormat,c)));
+
+        thresholdFailed.textProperty().addListener((obs, oldText, newText) -> {
+            if(handleTextChange(newText, originalAssetType.getValueFailed()))
+                assetType.setValueFailed(newText);
+        });
+        thresholdFailed.setTextFormatter( new TextFormatter<>(c -> UIUtilities.checkFormat(TextConstants.ThresholdValueFormat,c)));
+
+    }
+
+    /**
+     *  Handle the text change of the user fields to turn on or off the save functionality
+     *
+     * @author  Paul
+     */
+    private boolean handleTextChange(String newText, String field) {
+        if (field.equals(originalAssetType.getName()) || field.equals(originalAssetType.getDescription()) && newText.equals(field))
```

Comment:
Behaviour: when editing the description or the name, the save button stays disabled. Editing description is throwing the double.parse error while editing the name doesn't throw an exception, but doesn't enable the save button.
![Changing name_can't save](https://user-images.githubusercontent.com/47158909/105775854-86899100-5f35-11eb-9e76-78bcbbec807b.PNG)
![SetDescription_no save](https://user-images.githubusercontent.com/47158909/105775859-88ebeb00-5f35-11eb-840b-3f05bc9e8ef9.PNG)